smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpProtocolUnitTestRequestGenerator.kt

```diff
@@ -109,21 +112,41 @@ open class HttpProtocolUnitTestRequestGenerator protected constructor(builder: B
             operationMiddleware.renderMiddleware(ctx, writer, operation, operationStack, MiddlewareStep.DESERIALIZESTEP)
 
             if (ctx.settings.useInterceptors) {
+                val rpcService = serviceName
+                val rpcMethod = operation.getId().getName()
                 writer.write(
                     """
+                    var metricsAttributes = ${'$'}N()
+                    metricsAttributes.set(key: ${'$'}N.service, value: ${'$'}S)
+                    metricsAttributes.set(key: ${'$'}N.method, value: ${'$'}S)
```

Comment:
Similar to `MiddlewareExecutionGenerator`, generate the corresponding RPC service / method attributes.